framework/src/media/audio/audio_manager.c

```diff
@@ -871,31 +910,55 @@ int start_audio_stream_in(void *data, unsigned int frames)
 
 		medvdbg("Resume the input audio card!!\n");
 	}
-	
+
 	card->config[card->device_id].status = AUDIO_CARD_RUNNING;
 
+	if ((device_channel_num = pcm_get_channels(card->pcm)) == 0) {
+		meddbg("Fail to get channel number\n");
+		goto error_with_lock;
+	}
+	rechanneling_ratio = (float)card->resample.user_channel / (float)device_channel_num;
+
 	do {
 		// Todo: Logic needs to be changed.
 		//       From: 1 pcm_readi - 1 resampling
 		//       To  : * pcm_readi - 1 resampling
-		if (card->resample.necessary) {
+		if (card->resample.ratio != 1) {
 			unsigned int frames_to_read = (unsigned int)((float)frames / card->resample.ratio);
 			if (frames_to_read > (get_input_frame_count() / card->resample.ratio)) {
 				frames_to_read = (get_input_frame_count() / card->resample.ratio);
 			}
 
-			card->resample.buffer_size = get_input_frames_to_byte(frames);
+			card->resample.buffer_size = get_user_input_frames_to_byte(frames) / rechanneling_ratio;
 
 			ret = pcm_readi(card->pcm, card->resample.buffer, frames_to_read);
-			medvdbg("Read frames (%d/%u) for resample to %u\n", ret, frames_to_read, frames);
+			medvdbg("Read frames (%d/%u) in %u for resampling to %u\n", ret, frames_to_read, card->resample.buffer_size, frames);
 		} else {
-			ret = pcm_readi(card->pcm, data, frames);
+			if (rechanneling_ratio < 1) {	// Stereo -> Mono
+				int i = 0;
+				int len = card->resample.rechanneling_buffer_size / device_channel_num * rechanneling_ratio;
```

Comment:
i feel braces are needed as per the priority of operation & for code readability
what is your opinion

Reply:
In set_audio_stream_in(), rechanneling_buffer_size is determined based on frame size (int).
This calculation is just the reverse of it, so consideration to the priority of operation might not be needed.